cmd/integrationArtifactUpload.go

```diff
@@ -0,0 +1,218 @@
+	// Structs embedded in this way must each have a unique set of methods attached.
+	// If there is no struct which implements the method you need, attach the method to
+	// integrationArtifactUploadUtilsBundle and forward to the implementation of the dependency.
+}
+
+func newIntegrationArtifactUploadUtils() integrationArtifactUploadUtils {
+	utils := integrationArtifactUploadUtilsBundle{
+		Command: &command.Command{},
+	}
+	// Reroute command output to logging framework
+	utils.Stdout(log.Writer())
+	utils.Stderr(log.Writer())
+	return &utils
+}
+
+func integrationArtifactUpload(config integrationArtifactUploadOptions, telemetryData *telemetry.CustomData) {
+	// Utils can be used wherever the command.ExecRunner interface is expected.
+	// It can also be used for example as a mavenExecRunner.
+	httpClient := &piperhttp.Client{}
+
+	// For HTTP calls import  piperhttp "github.com/SAP/jenkins-library/pkg/http"
+	// and use a  &piperhttp.Client{} in a custom system
+	// Example: step checkmarxExecuteScan.go
+
+	// Error situations should be bubbled up until they reach the line below which will then stop execution
+	// through the log.Entry().Fatal() call leading to an os.Exit(1) in the end.
+	err := runIntegrationArtifactUpload(&config, telemetryData, httpClient)
+	if err != nil {
+		log.Entry().WithError(err).Fatal("step execution failed")
+	}
+}
+
+func runIntegrationArtifactUpload(config *integrationArtifactUploadOptions, telemetryData *telemetry.CustomData, httpClient piperhttp.Sender) error {
+	clientOptions := piperhttp.ClientOptions{}
+	httpClient.SetOptions(clientOptions)
+	header := make(http.Header)
+	header.Add("Accept", "application/json")
+	getIflowStatusURL := fmt.Sprintf("%s/api/v1/IntegrationDesigntimeArtifacts(Id='%s',Version='%s')", config.Host, config.IntegrationFlowID, config.IntegrationFlowVersion)
+	fmt.Printf(getIflowStatusURL)
+	tokenParameters := cpi.TokenParameters{TokenURL: config.OAuthTokenProviderURL, Username: config.Username, Password: config.Password, Client: httpClient}
+	token, err := cpi.CommonUtils.GetBearerToken(tokenParameters)
+	if err != nil {
+		return errors.Wrap(err, "failed to fetch Bearer Token")
+	}
+	clientOptions.Token = fmt.Sprintf("Bearer %s", token)
+	httpClient.SetOptions(clientOptions)
+	httpMethod := "GET"
+
+	//Check availability of integration artefact in CPI design time
+	iFlowStatusResp, httpErr := httpClient.SendRequest(httpMethod, getIflowStatusURL, nil, header, nil)
+
+	if iFlowStatusResp != nil && iFlowStatusResp.Body != nil {
+		defer iFlowStatusResp.Body.Close()
+	}
+
+	if iFlowStatusResp.StatusCode == 200 {
+		return UpdateIntegrationArtifact(config, httpClient)
+	} else if httpErr != nil && iFlowStatusResp.StatusCode == 404 {
+		return UploadIntegrationArtifact(config, httpClient)
+	}
+
+	if httpErr != nil {
+		return errors.Wrapf(httpErr, "HTTP %v request to %v failed with error", httpMethod, getIflowStatusURL)
```

Comment:
Wound't it be possible that the response body contains helpful hints for troubleshooting. Maybe we should try to get the response body into the log?

Reply:
this is resolved in latest commit